src/foam/util/UIDSupport.java

```diff
@@ -0,0 +1,151 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.util;
+
+public class UIDSupport {
+  /**
+   * Modulo constant for UID checksum
+   */
+  public final static int CHECKSUM_MOD = 997;
+
+  /**
+   * Singleton instance
+   */
+  private final static UIDSupport instance__ = new UIDSupport();
```

Comment:
Why does this need to be a singleton? It doesn't have any state and doesn't implement any interfaces. Couldn't it just be static methods?

Reply:
Updated https://github.com/kgrgreer/foam3/pull/765/commits/3802de6eeedf62ae6f3d7c2c930db8a4db36c6e3.